### PR TITLE
add rest of filter predicate tests

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues
a## Issues

[Predicates in filters can be cryptic](https://github.com/OpsLevel/team-platform/issues/433)
Second half of this [first predicate tests PR](https://github.com/OpsLevel/terraform-provider-opslevel/pull/424)

## Changelog

Add integration tests focusing on the following predicate filters:
- `aliases`
- `framework`
- `language`
- `lifecycle_index`
- `name`
- `product`
- `properties`
- `tags`
- `tier_index`

Note: `lifecycle_index` and `tier_index` are basically the same
`aliases`, `framework`, `language`, `name`, `product`, `properties` and `tags` are basically the same

- [X] List your changes here
- [ ] Make a `changie` entry, N/A testing only

## Tophatting

```bash
task test-integration -- \
 "-filter=filter_predicate_aliases.tftest.hcl" \
 "-filter=filter_predicate_framework.tftest.hcl" \
 "-filter=filter_predicate_language.tftest.hcl" \
 "-filter=filter_predicate_lifecycle_index.tftest.hcl" \
 "-filter=filter_predicate_name.tftest.hcl" \
 "-filter=filter_predicate_product.tftest.hcl" \
 "-filter=filter_predicate_properties.tftest.hcl" \
 "-filter=filter_predicate_tags.tftest.hcl" \
 "-filter=filter_predicate_tier_index.tftest.hcl"
```

```tf
task: [terraform-command] terraform -chdir=tests/remote test -var-file=test.tfvars -var="api_token=$OPSLEVEL_API_TOKEN" '-filter=filter_predicate_aliases.tftest.hcl' '-filter=filter_predicate_framework.tftest.hcl' '-filter=filter_predicate_language.tftest.hcl' '-filter=filter_p
redicate_lifecycle_index.tftest.hcl' '-filter=filter_predicate_name.tftest.hcl' '-filter=filter_predicate_product.tftest.hcl' '-filter=filter_predicate_properties.tftest.hcl' '-filter=filter_predicate_tags.tftest.hcl' '-filter=filter_predicate_tier_index.tftest.hcl'
filter_predicate_aliases.tftest.hcl... in progress
  run "resource_filter_with_aliases_predicate_contains"... pass
  run "resource_filter_with_aliases_predicate_equals"... pass
  run "resource_filter_with_aliases_predicate_exists"... pass
  run "resource_filter_with_aliases_predicate_matches_regex"... pass
  run "resource_filter_with_aliases_predicate_starts_or_ends_with"... pass
filter_predicate_aliases.tftest.hcl... tearing down
filter_predicate_aliases.tftest.hcl... pass
filter_predicate_framework.tftest.hcl... in progress
  run "resource_filter_with_framework_predicate_contains"... pass
  run "resource_filter_with_framework_predicate_equals"... pass
  run "resource_filter_with_framework_predicate_exists"... pass
  run "resource_filter_with_framework_predicate_matches_regex"... pass
  run "resource_filter_with_framework_predicate_starts_or_ends_with"... pass
filter_predicate_framework.tftest.hcl... tearing down
filter_predicate_framework.tftest.hcl... pass
filter_predicate_language.tftest.hcl... in progress
  run "resource_filter_with_language_predicate_contains"... pass
  run "resource_filter_with_language_predicate_equals"... pass
  run "resource_filter_with_language_predicate_exists"... pass
  run "resource_filter_with_language_predicate_matches_regex"... pass
  run "resource_filter_with_language_predicate_starts_or_ends_with"... pass
filter_predicate_language.tftest.hcl... tearing down
filter_predicate_language.tftest.hcl... pass
filter_predicate_lifecycle_index.tftest.hcl... in progress
  run "resource_filter_with_lifecycle_index_predicate_equals"... pass
  run "resource_filter_with_lifecycle_index_predicate_exists"... pass
  run "resource_filter_with_lifecycle_index_predicate_greater_than_or_equal_to"... pass
  run "resource_filter_with_lifecycle_index_predicate_less_than_or_equal_to"... pass
filter_predicate_lifecycle_index.tftest.hcl... tearing down
filter_predicate_lifecycle_index.tftest.hcl... pass
filter_predicate_name.tftest.hcl... in progress
  run "resource_filter_with_name_predicate_contains"... pass
  run "resource_filter_with_name_predicate_equals"... pass
  run "resource_filter_with_name_predicate_exists"... pass
  run "resource_filter_with_name_predicate_matches_regex"... pass
  run "resource_filter_with_name_predicate_starts_or_ends_with"... pass
filter_predicate_name.tftest.hcl... tearing down
filter_predicate_name.tftest.hcl... pass
filter_predicate_product.tftest.hcl... in progress
  run "resource_filter_with_product_predicate_contains"... pass
  run "resource_filter_with_product_predicate_equals"... pass
  run "resource_filter_with_product_predicate_exists"... pass
  run "resource_filter_with_product_predicate_matches_regex"... pass
  run "resource_filter_with_product_predicate_starts_or_ends_with"... pass
filter_predicate_product.tftest.hcl... tearing down
filter_predicate_product.tftest.hcl... pass
filter_predicate_properties.tftest.hcl... in progress
  run "resource_filter_with_properties_predicate_exists"... pass
  run "resource_filter_with_properties_predicate_satisfies_jq_expression"... pass
filter_predicate_properties.tftest.hcl... tearing down
filter_predicate_properties.tftest.hcl... pass
filter_predicate_tags.tftest.hcl... in progress
  run "resource_filter_with_tags_predicate_contains"... pass
  run "resource_filter_with_tags_predicate_equals"... pass
  run "resource_filter_with_tags_predicate_exists"... pass
  run "resource_filter_with_tags_predicate_matches_regex"... pass
  run "resource_filter_with_tags_predicate_starts_or_ends_with"... pass
  run "resource_filter_with_tags_predicate_satisfies_version_constraint"... pass
filter_predicate_tags.tftest.hcl... tearing down
filter_predicate_tags.tftest.hcl... pass
filter_predicate_tier_index.tftest.hcl... in progress
  run "resource_filter_with_tier_index_predicate_equals"... pass
  run "resource_filter_with_tier_index_predicate_exists"... pass
  run "resource_filter_with_tier_index_predicate_greater_than_or_equal_to"... pass
  run "resource_filter_with_tier_index_predicate_less_than_or_equal_to"... pass
filter_predicate_tier_index.tftest.hcl... tearing down
filter_predicate_tier_index.tftest.hcl... pass

Success! 41 passed, 0 failed.
```

## Changelog

- [ ] List your changes here
- [ ] Make a `changie` entry

### CODE REVIEW

File: tests/remote/filter_predicate_aliases.tftest.hcl [hunk 1]
```diff
@@ -0,0 +1,410 @@
+variables {
+  name            = "TF Test Filter with aliases predicate"
+  predicate_value = "fancy"
+  aliases_predicates = setproduct(
+    ["aliases"],
+    concat([
+      "contains",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_aliases.tftest.hcl:7]: Why do we have vars for `predicate_types_equals` and `predicate_types_exists` but not for these ones?

File: tests/remote/filter_predicate_aliases.tftest.hcl [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    error_message = format(
+      "expected predicate key 'aliases' got '%s'",
+      opslevel_filter.all_predicates["aliases_starts_with"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["aliases_starts_with"].predicate[0].type == "starts_with"
+    error_message = format(
+      "expected predicate type 'ends_with' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_aliases.tftest.hcl:391]:
```suggestion
      "expected predicate type 'starts_with' got '%s'",
```

File: tests/remote/filter_predicate_tier_index.tftest.hcl [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    error_message = format(
+      "expected predicate key 'tier_index' got '%s'",
+      opslevel_filter.all_predicates["tier_index_equals"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tier_index_equals"].predicate[0].type == "equals"
+    error_message = format(
+      "expected predicate type 'does_not_equal' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tier_index.tftest.hcl:73]:
```suggestion
      "expected predicate type 'equals' got '%s'",
```

File: tests/remote/filter_predicate_tags.tftest.hcl [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+    error_message = format(
+      "expected predicate key 'tags' got '%s'",
+      opslevel_filter.all_predicates["tags_contains"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tags_contains"].predicate[0].type == "contains"
+    error_message = format(
+      "expected predicate type 'does_not_contain' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tags.tftest.hcl:86]:
```suggestion
      "expected predicate type 'contains' got '%s'",
```

File: tests/remote/filter_predicate_tags.tftest.hcl [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+    error_message = format(
+      "expected predicate key 'tags' got '%s'",
+      opslevel_filter.all_predicates["tags_equals"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tags_equals"].predicate[0].type == "equals"
+    error_message = format(
+      "expected predicate type 'does_not_equal' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tags.tftest.hcl:175]:
```suggestion
      "expected predicate type 'equals' got '%s'",
```

File: tests/remote/filter_predicate_tags.tftest.hcl [hunk 6]
```diff
@@ -0,0 +1,512 @@
...
+    error_message = format(
+      "expected predicate key 'tags' got '%s'",
+      opslevel_filter.all_predicates["tags_exists"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tags_exists"].predicate[0].type == "exists"
+    error_message = format(
+      "expected predicate type 'does_not_exist' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tags.tftest.hcl:260]:
```suggestion
      "expected predicate type 'exists' got '%s'",
```

File: tests/remote/filter_predicate_tags.tftest.hcl [hunk 7]
```diff
@@ -0,0 +1,512 @@
...
+    error_message = format(
+      "expected predicate key 'tags' got '%s'",
+      opslevel_filter.all_predicates["tags_matches_regex"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tags_matches_regex"].predicate[0].type == "matches_regex"
+    error_message = format(
+      "expected predicate type 'does_not_match_regex' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tags.tftest.hcl:345]:
```suggestion
      "expected predicate type 'matches_regex' got '%s'",
```

File: tests/remote/filter_predicate_tags.tftest.hcl [hunk 8]
```diff
@@ -0,0 +1,512 @@
...
+    error_message = format(
+      "expected predicate key 'tags' got '%s'",
+      opslevel_filter.all_predicates["tags_starts_with"].predicate[0].key
+    )
+  }
+
+  assert {
+    condition = opslevel_filter.all_predicates["tags_starts_with"].predicate[0].type == "starts_with"
+    error_message = format(
+      "expected predicate type 'ends_with' got '%s'",
```
Comment (taimoorgit) [inline tests/remote/filter_predicate_tags.tftest.hcl:434]:
```suggestion
      "expected predicate type 'starts_with' got '%s'",
```

**Review by taimoorgit — CHANGES_REQUESTED** [review]
Was not able to check all tftest.hcl files, there are error messages that need corrections

File: tests/remote/filter_predicate_aliases.tftest.hcl (hunk 1 above)
Comment (davidbloss) [reply tests/remote/filter_predicate_aliases.tftest.hcl:7]: I just never came back around to making them but yeah these can be added

**Review by taimoorgit — APPROVED** [review]

**Pull request merged**